internal/driver/glfw/menu_bar_item.go

```diff
@@ -113,6 +113,11 @@ func (i *menuBarItem) MouseOut() {
 // Implements: fyne.Tappable
 func (i *menuBarItem) Tapped(*fyne.PointEvent) {
 	i.Parent.toggle(i)
+
+	if i.Parent.active {
+		i.Parent.canvas.Focus(i)
+	}
+	i.Refresh()
```

Comment:
There is no test for this, therefore I don’t know which bug you’re trying to fix with it.
But I think this is partly wrong. The parent’s `toggle` already focuses the item on activation and unfocuses it on deactivation.
So, if the menu is active after the toggle, the item should already be focused and the added focus code does nothing.
If the additional `Refresh` fixes some bug, we should try to add a test case for it and also move it to `MenuBar`’s `toggle()`. Currently, the item gets refreshed before it is being focused. Maybe that’s to early?

Reply:
Apologies it was not a bug fix, I had misread the code in the refactor and you are right it should not have been needed. What has changed is that on tapped the focus in is not triggering so we don't get the same refresh request cycle. I tracked down that `activateChild` does not actually refresh the button - only the bar. The bar's refresh does not cascade to the list of items it would seem. I have put in place what feels like a better fix in `activateChild`, what do you think?